sdk/python/kfp/dsl/databricks/_job_op.py

```diff
@@ -0,0 +1,169 @@
+#       status.last_10_runs: [type Run struct](https://github.com/xinsnake/databricks-sdk-golang/blob/master/azure/models/Run.go)
+# Databricks Rest API 2.0 Data Structures:
+#   [Jobs API Data Structures](https://docs.databricks.com/dev-tools/api/latest/jobs.html#data-structures) 
+#   [Clusters API Data Structures](https://docs.databricks.com/dev-tools/api/latest/clusters.html#data-structures)
+
+class CreateJobOp(ResourceOp):
+    """Represents an op which will be translated into an Azure Databricks Spark Job creation resource template.
+
+    Example:
+
+        import kfp.dsl.databricks as databricks
+
+        databricks.CreateJobOp(
+            name="createjob",
+            job_name_prefix="test-job-",
+            spec={
+                "new_cluster" : {
+                    "spark_version":"5.3.x-scala2.11",
+                    "node_type_id": "Standard_D3_v2",
+                    "num_workers": 2
+                },
+                "libraries" : [
+                    {
+                        "jar": 'dbfs:/my-jar.jar'
+                    },
+                    {
+                        "maven": {
+                            "coordinates": 'org.jsoup:jsoup:1.7.2'
+                        }
+                    }
+                ],
+                "timeout_seconds" : 3600,
+                "max_retries": 1,
+                "schedule":{
+                    "quartz_cron_expression": "0 15 22 ? * *",
+                    "timezone_id": "America/Los_Angeles",
+                },
+                "spark_jar_task": {
+                    "main_class_name": "com.databricks.ComputeModels",
+                },
+            }
+        )
+    """
+
+    def __init__(self,
+        name: str = None,
+        job_name_prefix: str = None,
+        spec=None):
+        """Create a new instance of CreateJobOp.
+
+        Args:
+
+            name: the name of the op. It does not have to be unique within a pipeline
+                because the pipeline will generates a unique new name in case of conflicts.
+            job_name_prefix: the prefix of the name of the Spark Job. A sufix will be added to 
+                generate a unique name for the job.
+            spec: Specification of the Spark job to create.
+
+        Raises:
+            ValueError: if not inside a pipeline
+                        if the name is an invalid string
+                        if no job name prefix is provided
+                        if no spec is provided
+        """
+
+        if not job_name_prefix:
+            raise ValueError("You need to provide a job_name_prefix.")
+
+        if not spec:
+            raise ValueError("You need to provide a spec.")
+
+        super().__init__(
+            k8s_resource = {
+                "apiVersion": "databricks.microsoft.com/v1alpha1",
+                "kind": "Djob",
+                "metadata": {
+                    "name": str(job_name_prefix) + str(time.time()),
```

Comment:
why are we appending the time? I understand that job names have to unique, but we don't follow this pattern in the cluster DSL

Reply:
Didn't focus too much on Jobs yet. I just got that code from an example pipeline that was using ResourceOp directly. But you are right, we can remove that for consistency.